pulp_smart_proxy/app/views.py

```diff
@@ -0,0 +1,73 @@
+from urllib.parse import urljoin
+
+from django.conf import settings
+from drf_spectacular.utils import extend_schema
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from pulpcore.app.apps import pulp_plugin_configs, get_plugin_config
+
+
+class FeaturesView(APIView):
+    """
+    Returns features of the smart_proxy
+    """
+
+    @extend_schema(
+        summary="Inspect features",
+        operation_id="features_read",
+    )
+    def get(self, request):
+        data = ["pulpcore"]
+        return Response(data)
+
+
+class FeaturesV2View(APIView):
+    """
+    Returns features of the smart_proxy in v2 format
+    """
+
+    @extend_schema(
+        summary="Inspect features",
+        operation_id="featuresv2_read",
+    )
+    def get(self, request):
+        # there is no setting for the API url
+        # not adding /pulp/api/v3 here as Katello does so on its own
+        pulp_url = request.build_absolute_uri("/")
```

Comment:
interestingly, this ends up being `http://…`, not `https://` :thinking:

Reply:
Are the headers correctly forwarded and trusted? We have https://github.com/theforeman/puppet-pulpcore/blob/8288f3736a1543417eba6322e7013d0f230bd85e/templates/settings.py.erb#L65-L66 in our installer, together with some Apache logic. In this repo I do see `RequestHeader set X_FORWARDED_PROTO "https"` but not `SECURE_PROXY_SSL_HEADER` in `settings.py`.

Reply:
https://github.com/theforeman/puppet-foreman/pull/1214

Reply:
Technically we should enforce the same headers in puppet-pulpcore

Reply:
https://github.com/theforeman/puppet-pulpcore/blob/8288f3736a1543417eba6322e7013d0f230bd85e/templates/settings.py.erb#L65-L66 uses that, w/o actually setting it anywhere.

Reply:
So my whole intent to get it right back then was all meaningless. Looks like I never verified it and we never relied on it.